### PR TITLE
ENG-3585 Add friendly code duplicates validation across different languages

### PR DESCRIPTION
**Pull request by alepintus**

### CODE REVIEW

File: src/main/java/org/entando/entando/plugins/jpseo/web/page/SeoPageController.java [hunk 1]
```diff
@@ -105,10 +105,15 @@ public ResponseEntity<SimpleRestResponse<SeoPageDto>> addPage(UserDetails user,
                 }
             }
         }
-        if (bindingResult.hasErrors()) {
```
Comment (zonia3000) [inline src/main/java/org/entando/entando/plugins/jpseo/web/page/SeoPageController.java:108]: This block should stay outside the `if (null!=pageRequest.getSeoData()){}` block.
Indeed, if API is called using an already existing page code, the previous line `getSeoPageValidator().validate(pageRequest, bindingResult)` detects the problem and add a proper error to the `bindingResult` object. However, if this block is moved into that if and the payload has null `seoData` a `ValidationConflictException` will never be thrown.

File: src/main/java/org/entando/entando/plugins/jpseo/web/page/validator/SeoPageValidator.java [hunk 2]
```diff
@@ -58,5 +63,28 @@ public boolean checkFriendlyCode(String pageCode, String friendlyCode) {
         return true;
     }
 
+    public boolean validateFriendlyCodeByLang(Map<String, SeoDataByLang> seoDataByLang, Errors errors) {
+
+        List friendlyCodes = new ArrayList<String>();
+        seoDataByLang.forEach(
+                (key, value) -> {
+                    if (null != value.getFriendlyCode()) {
```
Comment (zonia3000) [inline src/main/java/org/entando/entando/plugins/jpseo/web/page/validator/SeoPageValidator.java]: Ignore also empty friendly codes. By default app-builder sends empty friendly codes, so an error would be triggered for pages having no friendly codes defined.

**Review by zonia3000 — CHANGES_REQUESTED** [review]

File: src/main/java/org/entando/entando/plugins/jpseo/web/page/SeoPageController.java (hunk 1 above)
Comment (alepintus) [reply src/main/java/org/entando/entando/plugins/jpseo/web/page/SeoPageController.java:108]: Done

File: src/main/java/org/entando/entando/plugins/jpseo/web/page/validator/SeoPageValidator.java (hunk 2 above)
Comment (alepintus) [reply src/main/java/org/entando/entando/plugins/jpseo/web/page/validator/SeoPageValidator.java]: Done

**Review by zonia3000 — APPROVED** [review]

**Pull request merged**